assert/http_assertions.go

```diff
@@ -163,3 +164,50 @@ func HTTPBodyNotContains(t TestingT, handler http.HandlerFunc, method, url strin
 
 	return !contains
 }
+
+// HTTP asserts that a specfied handler returns set of expected values given by HttpOptions.
+//
+//	assert.HTTP(t, myHandler, "www.google.com", nil, WithCode(200), WithBody("I'm Feeling Lucky"), WithRequestHeader(http.Header{"a": []string{"b"}}, WithExpectedBody(bytes.NewBuffer("c"))))
+//
+// Returns whether the assertion was successful (true) or not (false).
+func HTTP(t TestingT, handler http.HandlerFunc, method, url string, values url.Values, options ...HttpOption) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+
+	b := &builder{}
```

Comment:
Since the `options` is optional, should the builder have some defaults?

Consider this:

```go
func handler(w http.ResponseWriter, r *http.Request) {
	w.Write([]byte("hello"))
}

func TestHttp(t *testing.T) {
	assert.HTTP(t, handler, http.MethodGet, "/", nil)
}
```

Fails with:

```
=== RUN   TestHttp
    main_test.go:15: 
        	Error Trace:	/Users/armahishi/dev/personal/forks/testify/example/main_test.go:15
        	Error:      	Expected HTTP success status code for "/?" but received 200
        	Test:       	TestHttp
--- FAIL: TestHttp (0.00s)
FAIL
FAIL	github.com/stretchr/testify/example	0.173s
FAIL
```

Reply:
Good point I have added the default code for builder.

Reply:
I have made a change, based on what you said in next review step.
There are no defaults as the option would override something that is default for the builder. You mentioned that you expected that the `status.code` is not tested as You did not ask for it.
As the options should be optional, I made a change when you select no option, there is nothing to be tested so the test fails. It is possible to to remove this check, however it is incorrectly tested `handler` as there is not `assertion`.